### DISCUSSION TITLE
Self-Hosted Database

### DISCUSSION THREAD

**Discussion by ErickIriasM** [AppFlowy-IO/AppFlowy#2943] [General]
Hi,

I was looking the documentation but I couldn't see it though some websites say it so. I want to ask before try explore deeper.

I read some discussions regarding this, but I think start it fresh would be good.

Can I deploy this as a self-hosted app and access it through web browser or the installed app?

Or do something like this?

Expectation: Install this in my computer to host and use my other computers/devices access it through browser or app but connecting the same database. Also adding my family's devices with their respective usernames (as a Team)

**Comment by annieappflowy** [comment]
Hi @ErickIriasM , self host is under development. Once it's done, you will be able to connect to the same database from AppFlowy across different devices.

**Reply by njakes** [reply]
Is there a roadmap or something we could follow for the self hosted server side? Or do you plan on having a release announcement when it is complete?

**Reply by arsen-z** [reply]
+1 this, also very interested when self-hosting would be available - would work great for our laboratory projects!

**Reply by annieappflowy** [reply]
https://docs.appflowy.io/docs/essential-documentation/appflowy-with-supabase
AppFlowy can now be self-hosted with Supabase. You can find a step-by-step in the linked article.
We're working on self-hosting with Docker. It's expected to release next month.

**Reply by Zagrthos** [reply]
> https://docs.appflowy.io/docs/essential-documentation/appflowy-with-supabase AppFlowy can now be self-hosted with Supabase. You can find a step-by-step in the linked article. We're working on self-hosting with Docker. It's expected to release next month.

Quick question about this:
Can't we just deploy the AppFlowy Cloud App or is it something completely different?
I'm thinking about using this as a primary documentation platform and I want it to host on a server, where I can then use the Web Interface to communicate with it (Or did I understand something wrong and this is just a client?)

**Reply by martin-braun** [reply]
@annieappflowy This is kinda cool, but if I get this right, you essentially host the entire service on your own server, thus user can create accounts and login to their own personal space? What we are especially interested in is to host a mono space for many users that cannot register, but need to be registered beforehand. Also it would be cool if AppFlowy could run in the browser, thus hosting the Rust backend on the server behind a REST client.

AppFlowy looks promising and has potential, but I can't say it enough, a web first approach would've been very useful, because it seems I have to publish macOS and Windows clients with my own env variables to make it work.

**Reply by annieappflowy** [reply]
https://docs.appflowy.io/docs/guides/appflowy/self-hosting-appflowy
@ErickIriasM @arsen-z @Zagrthos @njakes @martin-braun

**Comment by monitor403** [comment]
> https://docs.appflowy.io/docs/essential-documentation/appflowy-with-supabase AppFlowy can now be self-hosted with Supabase. You can find a step-by-step in the linked article. We're working on self-hosting with Docker. It's expected to release next month.

Hi, I would like to ask when self-hosting with Docker will be released? Is there a plan to release it later?

**Reply by marcusze** [reply]
Same question here

**Reply by jelmerveen** [reply]
Would also love a self-hosting version for linux (non-docker), would that be possible?

**Reply by ChrisClems** [reply]
I came here researching this exact question. It says it was expected to release "Next month" back in September.

**Reply by njakes** [reply]
I would also like to know if this is in the works. Thank you for your efforts in creating this amazing software.

**Reply by abdessalaam** [reply]
Me too! Thanks for your great work.

**Reply by annieappflowy** [reply]
Check out our guide https://docs.appflowy.io/docs/guides/appflowy/self-hosting-appflowy to host AppFlowy with cloud services such as Amazon EC2 and Azure Virtual Machines. Let us know if you need help! @marcusze @jelmerveen @neilclemons @njakes @bpawel-bclub @monitor403

**Comment by annieappflowy** [comment]
Hello everyone on this thread, we're thrilled to introduce self-hosting AppFlowy with AppFlowy Cloud, designed for easy deployment and self-managed cloud storage. It can be hosted with cloud services such as Amazon EC2 and Azure Virtual Machines. For more information and setup instructions, please check out our guide: https://docs.appflowy.io/docs/guides/appflowy/self-hosting-appflowy. We welcome your feedback!

**Reply by stefanuswiely** [reply]
Hello @annieappflowy,  I am successfully deploy the self-hosted appflowy use the guide at DigitalOcean droplets service. I need to do manual backup for the data, there is any guide how to backup the data? Thank you.

**Reply by khorshuheng** [reply]
Data on AppFlowy is stored in both Postgres and S3. Hence, as long as the data on these storage are back up, you can restore the original data.

We also have the option to export documents in different format, in case you want to keep backup by importing them into different tools.
